src/main/java/org/folio/helper/PoNumberHelper.java

```diff
@@ -42,6 +48,20 @@ public Future<Void> checkPONumberUnique(PoNumber poNumber, RequestContext reques
     return promise.future();
   }
 
+  public Future<Void> checkPONumberUnique(String poNumber, RequestContext requestContext) {
+    if (StringUtils.isEmpty(poNumber)){
+      return Future.succeededFuture();
+    }
+    return purchaseOrderStorageService.getPurchaseOrderByPONumber(poNumber, requestContext)
+      .map(po -> {
+        if (po.getInteger("totalRecords") != 0) {
+          logger.error("Exception validating PO Number existence");
+          throw new HttpException(400, ErrorCodes.PO_NUMBER_ALREADY_EXISTS);
+        }
+        return null;
+      });
+  }
+
   public Future<PoNumber> getPoNumber(RequestContext requestContext) {
```

Comment:
Is not used, can be deleted

Reply:
`PoNumberAPI::getOrdersPoNumber` uses it. Can you double check?